### ISSUE TITLE
ReadOnlyMemoryError() and EXCEPTION_ACCESS_VIOLATION errors when running distributed processes on Windows

### ISSUE THREAD

**Issue by martinbiel** [JuliaLang/julia#40766]
My distributed CI [tests](https://github.com/martinbiel/StochasticPrograms.jl/tree/master/test/distributed) are consistently failing on Windows 1.6/nightly. I do not always get logs since Github actions breaks, but from those I get it seems to stem from ReadOnlyMemoryError() and EXCEPTION_ACCESS_VIOLATION issues. Because the same tests work fine on Linux/Mac I am guessing that it is a Windows problem rather than something from my tests.

I do not use Windows so it is difficult to produce a MWE. I am attaching some timestamped logs from the Github actions runs:
[error_windows_1_6_210430.txt](https://github.com/JuliaLang/julia/files/6450433/error_windows_1_6_210430.txt)
[error_windows_1_6_210505.txt](https://github.com/JuliaLang/julia/files/6450434/error_windows_1_6_210505.txt)
[error_windows_1_6_210508.txt](https://github.com/JuliaLang/julia/files/6450435/error_windows_1_6_210508.txt)
[error_windows_nightly_210430.txt](https://github.com/JuliaLang/julia/files/6450436/error_windows_nightly_210430.txt)
[error_windows_nightly_210508.txt](https://github.com/JuliaLang/julia/files/6450437/error_windows_nightly_210508.txt)
Notably, the tests do not always crash at the same place, but always after starting a distributed environment with `addprocs`.

If the errors are difficult to reproduce or do appear related to my package, I could take a shot at producing a better MWE, but I figured I would try to post the raw logs first.

**Comment by Seelengrab** [comment]
I can't reproduce the failure you're describing on my machine (windows 10), but your tests seem to fail for a different reason:

```
WARNING: both Distributions and JuMP export "mode"; uses of it in module StochasticPrograms must be qualified
WARNING: both Distributions and JuMP export "shape"; uses of it in module StochasticPrograms must be qualified
test_all_decision_variables: Error During Test at C:\Users\valentin\.julia\packages\StochasticPrograms\54439\test\decisions\decision_variable.jl:496
  Got exception outside of a @test
  MethodError: Cannot `convert` an object of type
    OrderedCollections.OrderedDict{MathOptInterface.ConstraintIndex{SingleDecision, StochasticPrograms.FreeDecision}, Tuple{SingleDecision, StochasticPrograms.FreeDecision}} to an object of type
    MathOptInterface.Utilities.VectorOfConstraints
```

at `src\types\decisions\expressions\affine.jl:71`.

I've just done `]add StochasticPrograms` followed by `]test StochasticPrograms`, so maybe this was a wrong version?

**Comment by martinbiel** [comment]
Yes, the first one stalled for 5 hours (the tests typically take 30 minutes), so the InterruptException is most likely the CI shutting it down after reaching a time limit. The latest one (3rd file) crashed after 19 minutes and does not show an InterruptException.

The Windows tests do not fail on Julia 1.0 and also did not fail on 1.5 when I still tested against it. So my best guess would be that 1.6 introduced the possibility for these memory issues.

**Comment by vtjnash** [comment]
I think that is https://github.com/JuliaLang/Pkg.jl/issues/2323 haha

**Comment by Seelengrab** [comment]
@martinbiel master seems to work on my machine on 1.6.1, so probably a CI specific issue, not a julialang one.

**Comment by martinbiel** [comment]
All right, good to know. Thanks for testing it!

**Comment by martinbiel** [comment]
In light of the above, I have disabled distributed CI tests on Windows. 

I guess this issue does not have to be tracked?

**Comment by Seelengrab** [comment]
At least not on the julia level, no - though I guess you could open an issue on your package repo, for the future?
